File: frontend/console/src/api/fsu.js
import Axios from "@/http";

export function createFsu(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/fsu', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveFsu(fsuId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/fsu/${ fsuId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateFsu(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/fsu', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteFsu(fsuId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/fsu/${ fsuId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteFsuWithFlag(fsuId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/fsu/${ fsuId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllFsus() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/fsu/all').then(res => {
            resolve(res)
        })
    })
}

export function queryFsu (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/fsu/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/fsu.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_FSU = "/config/fsu"
const URL_ROUTER_FSU_BY_ID = "/config/fsu/:fsu-id"
const URL_ROUTER_FSU_QUERY = "/config/fsu/query"
const URL_ROUTER_FSU_ALL = "/config/fsu/all"

func FsuRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_FSU, config.CreateFsu)
	r.GET(URL_ROUTER_FSU_BY_ID, config.RetrieveFsu)
	r.PUT(URL_ROUTER_FSU, config.UpdateFsu)
	r.DELETE(URL_ROUTER_FSU_BY_ID, config.DeleteFsu)
	r.PUT(URL_ROUTER_FSU_BY_ID, config.DeleteFsuWithFlag)
	r.POST(URL_ROUTER_FSU_QUERY, config.QueryFsu)
	r.GET(URL_ROUTER_FSU_ALL, config.GetAllFsus)
	return r
}